src/test/activation/activationService.unit.test.ts

```diff
@@ -578,33 +578,52 @@ suite('Activation - ActivationService', () => {
 
         test('Track current LS usage for first usage', async () => {
             state.reset();
-            state.setup(s => s.value).returns(() => undefined).verifiable(TypeMoq.Times.once());
-            state.setup(s => s.updateValue(TypeMoq.It.isValue(true))).returns(() => Promise.resolve()).verifiable(TypeMoq.Times.once());
+            state.setup(s => s.value).returns(() => undefined).verifiable(TypeMoq.Times.exactly(2));
+            state.setup(s => s.updateValue(TypeMoq.It.isValue(true))).returns(() => {
+                state.setup(s => s.value).returns(() => true);
+                return Promise.resolve();
+            }).verifiable(TypeMoq.Times.once());
 
             const activationService = new LanguageServerExtensionActivationService(serviceContainer.object, stateFactory.object, experiments.object);
-            await activationService.trackLangaugeServerSwitch(true);
+            await activationService.sendTelemetryForChosenLanguageServer(true);
 
             state.verifyAll();
         });
         test('Track switch to LS', async () => {
             state.reset();
-            state.setup(s => s.value).returns(() => true).verifiable(TypeMoq.Times.once());
+            state.setup(s => s.value).returns(() => true).verifiable(TypeMoq.Times.exactly(2));
             state.setup(s => s.updateValue(TypeMoq.It.isValue(false))).returns(() => Promise.resolve()).verifiable(TypeMoq.Times.once());
```

Comment:
It seems silly not to make this more readable too:

```suggestion
            state.setup(s => s.updateValue(TypeMoq.It.isValue(false)))
                .returns(() => Promise.resolve())
                .verifiable(TypeMoq.Times.once());
```